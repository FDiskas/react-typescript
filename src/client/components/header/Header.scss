.header {
    background-color: #ffefc8;
}
