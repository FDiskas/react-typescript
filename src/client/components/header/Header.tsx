import * as React from 'react';

import * as styles from './Header.scss';

export interface IHeaderProps {
}

export interface IHeaderState {
}

export class Header extends React.Component<IHeaderProps, IHeaderState> {
    constructor(props: IHeaderProps) {
        super(props);
    }

    render() {
        return (
            <div className={styles.header}>
                <h1>App</h1>
                Component name: Header
                File name: Header.tsx
            </div>
        );
    }
}
