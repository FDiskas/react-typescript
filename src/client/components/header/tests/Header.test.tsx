import * as React from 'react';
import { shallow } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';

import { Header } from '../Header';

describe('Header.js', () => {
    it('renders without crashing', () => {
        shallow(<Header/>);
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<Header/>);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });
});
