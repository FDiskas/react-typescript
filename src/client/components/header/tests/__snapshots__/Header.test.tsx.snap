// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.js matches snapshot 1`] = `
<div
  className="header"
>
  <h1>
    App
  </h1>
  Component name: Header File name: Header.tsx
</div>
`;
